cairo-contracts/packages/clients/src/cometbft/component.cairo

```diff
@@ -415,6 +440,26 @@ pub mod CometClientComponent {
             self.client_sequence.read()
         }
 
+        /// Returns the heights at which the client has been updated in
+        /// descending order.
+        fn read_update_heights(
+            self: @ComponentState<TContractState>, client_sequence: u64
+        ) -> Array<Height> {
+            let mut heights: Array<Height> = ArrayTrait::new();
+
+            let entry = self.update_heights.entry(client_sequence);
```

Comment:
Can't we just return the entry here as `Vec<Height>`?

Reply:
Not actually. Getting `Vec<Height>` is possible only if we could call `read()` instead of `entry()`, though this method is not available on `Map` containing a `Vec`. 
The `entry` in fact is of type `StoragePath<Vec<Height>>` which gives access to some methods on `Vec`, but not allowing to return the own `Vec`.

Reply:
Something like `filter` or `filter_map` could do that. But a loop is fine for now.

Reply:
I see. That's unfortunate. There isn't any sort of fliter-like methods available.